api/validators.py

```diff
@@ -0,0 +1,152 @@
+import os
+import copy
+import json
+import logging
+import jsonschema
+
+log = logging.getLogger('scitran.api')
+# following https://github.com/Julian/jsonschema/issues/98
+# json schema files are expected to be in the schemas folder relative to this module
+schema_path = os.path.abspath(os.path.dirname(__file__))
+
+resolver = jsonschema.RefResolver('file://' + schema_path + '/schemas/', None)
+
+expected_mongo_schemas = set([
+    'acquisition.json',
+    'collection.json',
+    'container.json',
+    'file.json',
+    'group.json',
+    'note.json',
+    'permission.json',
+    'project.json',
+    'session.json',
+    'subject.json',
+    'user.json',
+    'avatars.json',
+    'tag.json'
+])
+expected_input_schemas = set([
+    'acquisition.json',
+    'collection.json',
+    'container.json',
+    'file.json',
+    'group.json',
+    'note.json',
+    'permission.json',
+    'project.json',
+    'session.json',
+    'subject.json',
+    'user.json',
+    'avatars.json',
+    'download.json',
+    'tag.json'
+])
+mongo_schemas = set()
+input_schemas = set()
+# validate and cache schemas at start time
+for schema_file in os.listdir(schema_path + '/schemas/mongo/'):
+    mongo_schemas.add(schema_file)
+    resolver.resolve('mongo/' + schema_file)
+
+assert mongo_schemas == expected_mongo_schemas, '{} is different from {}'.format(mongo_schemas, expected_schemas)
```

Comment:
This allow to check at API start time that all the needed schemas (and only these) are present.